app/controllers/application_controller.rb

```diff
@@ -1,10 +1,150 @@
 require './config/environment'
+require 'bcrypt'
 
 class ApplicationController < Sinatra::Base
 
   configure do
+    enable :sessions
+    set :session_secret, 'password_security'
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if session[:id]
+      redirect '/tweets'
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+    user = User.new(username: params[:username], email: params[:email], password: params[:password])
+    if user.save
+      user.save
+      session[:id] = user.id
+      redirect '/tweets'
+    else
+      redirect '/signup'
+    end
+  end
+
+  get '/login' do
+    if session[:id]
+      redirect '/tweets'
+    end
+    erb :'/users/login'
+  end
+
+  post '/login' do
+    user = User.find_by(username: params[:username])
+    if logged_in?
+      redirect '/tweets'
+    elsif user.authenticate(params[:password])
+      session[:id] = user.id
+      redirect '/tweets'
+    end
+  end
+
+  get '/tweets' do
+    if session[:id]
+      @user = User.find(session[:id])
+      erb :'/tweets/tweets'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets/new' do
+    if !logged_in?
```

Comment:
Typically, tweet creation would be posted to `/tweets` instead of `/tweets/new`.